techmatters/flex-plugins
# CHI-1617 Github action for terraform plan and role based terraform

Primary reviewer:

## Description
This sets terraform up for role based privilege escalation and adds a github workflow to run terraform plan.

This depends on techmatters/infrastructure-config#105

This should be merged after #1151 and #1153
This should be merged before #1144

### Checklist

### Related Issues
Fixes #CHI-1617

### Verification steps


See test run here: https://github.com/techmatters/flex-plugins/actions/runs/3795028591 The `plan.log` artifact is the human readable diff. Secrets should be properly marked as sensitive since they are pulled from SSM and TF will prevent their accidental display.
